src/main/java/vazkii/botania/common/crafting/StateIngredientHelper.java

```diff
@@ -91,6 +133,42 @@ public static StateIngredient read(PacketBuffer buffer) {
 		}
 	}
 
+	/**
+	 * Resolves tag ingredients, returning null if their tag doesn't exist, then returns a new ingredient
+	 * if the replacement function returns a replacement, used to filter ores deprioritized with the config.
+	 */
+	@Nullable
+	public static StateIngredient resolveAndFilter(StateIngredient ingredient, @Nonnull UnaryOperator<List<Block>> function) {
```

Comment:
maybe make this an instance method on StateIngredient instead of a static one with if-elses?